### PR TITLE
Add secrets controller

### PR DESCRIPTION
**Pull request by csrwng**
Now instantiates jenkins if not present

### CODE REVIEW

**Comment by csrwng** [comment]
@stevekuznetsov ptal

**Comment by csrwng** [comment]
related artifacts can be instantiated with jenkins/controllers/secrets/jenkins-secrets-controller.yaml

File: jenkins/controllers/secrets/bin/run.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	oc new-app --template="${JENKINS_TEMPLATE_NAME}"
+	while [[ $(oc get endpoints jenkins -o jsonpath='{ .subsets[*].addresses[0].ip }') == "" ]]; do
+		sleep 1
+	done
+fi
+
+oc observe secrets                                                          \
+   --names=secret-names.py                                                  \
+   --delete=secret-delete.py                                                \
+   -a "{ .metadata.annotations['ci\\.openshift\\.io/jenkins-secret-id'] }"  \
```
Comment (stevekuznetsov) [inline jenkins/controllers/secrets/bin/run.sh]: Did we check if the simpler syntax worked here?

Comment (csrwng) [inline jenkins/controllers/secrets/bin/run.sh]: I did and it doesn't work

**Comment by csrwng** [comment]
added the simplified jsonpath again

**Pull request merged**